src/Setting/SettingsController.php

```diff
@@ -101,6 +103,63 @@ public function validateApiCredentials(Request $request): JsonResponse
         return new JsonResponse(['credentialsValid' => $credentialsValid]);
     }
 
+    #[OA\Get(
+        path: '/api/_action/paypal/test-api-credentials',
+        operationId: 'testApiCredentials',
+        tags: ['Admin Api', 'PayPal'],
+        responses: [new OA\Response(
+            response: Response::HTTP_OK,
+            description: 'Returns if the provided API credentials are valid',
+            content: new OA\JsonContent(
+                required: ['valid', 'errors'],
+                properties: [
+                    new OA\Property(
+                        property: 'valid',
+                        type: 'boolean',
+                    ),
+                    new OA\Property(
+                        property: 'errors',
+                        type: 'array',
+                        items: new OA\Items(ref: '#/components/schemas/error'),
+                    ),
+                ]
+            )
+        )]
+    )]
+    #[Route(path: '/api/_action/paypal/test-api-credentials', name: 'api.action.paypal.test-api-credentials', methods: ['POST'], defaults: ['_acl' => ['swag_paypal.viewer']])]
+    public function testApiCredentials(RequestDataBag $data): JsonResponse
+    {
+        $clientId = $data->getString('clientId');
+        if (!$clientId) {
+            throw RoutingException::invalidRequestParameter('clientId');
+        }
+
+        $clientSecret = $data->getString('clientSecret');
+        if (!$clientSecret) {
+            throw RoutingException::invalidRequestParameter('clientSecret');
+        }
+
+        $merchantPayerId = $data->get('merchantPayerId');
+        if ($merchantPayerId !== null && !\is_string($merchantPayerId)) {
+            throw RoutingException::invalidRequestParameter('merchantPayerId');
+        }
+
+        $sandboxActive = $data->getBoolean('sandboxActive');
+
+        try {
+            /* @phpstan-ignore-next-line method will have additional method */
+            $valid = $this->apiCredentialService->testApiCredentials($clientId, $clientSecret, $sandboxActive, $merchantPayerId);
+        } catch (PayPalApiException $error) {
+            $valid = false;
+            $errors = (new ErrorResponseFactory())->getErrorsFromException($error);
+        }
+
+        return new JsonResponse([
+            'valid' => $valid,
+            'errors' => $errors ?? [],
+        ]);
+    }
```

Comment:
can't we keep the existing endpoint and change the response based on the FF?

Reply:
it's rewritten to be a post endpoint, instead of a get-endpoint where all creds are passed via query params